core/schema/container.graphqls

```diff
@@ -668,6 +668,33 @@ type Container {
     mediaTypes: ImageMediaTypes = OCIMediaTypes
   ): Boolean!
 
+  """
+  Returns a File representing the container serialized to a tarball.
+  """
+  asTarball(
+    """
+    Identifiers for other platform specific containers.
+    Used for multi-platform image.
+    """
+    platformVariants: [ContainerID!]
```

Comment:
Agreed, the multiplatform support was so long ago that I forget the exact reasoning for this exact approach, though I remember it being a practical compromise for some reason or another rather than "this is the ideal api". Would need to dig that PR up.

I have some ideas on a total re-thinking of all the publish/export/etc. related APIs in general too, so yes I agree better to just stick with this for now and save for a future wider-scale refactoring.